### PR TITLE
[Merged by Bors] - feat: port Algebra.Category.Module.Subobject

### PR DESCRIPTION
**Pull request by nick-kuhn**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by nick-kuhn** [comment]
What is the Standard regarding naming here? Since this is the module category and was capitalized in Lean3, should this be `ModuleCat` now? It seems like that is what's happening for other files.

**Comment by nick-kuhn** [comment]
This is basically ready, except for two issues:
1. I manually moved the file from the Module to ModuleCat folder, since this seemed to agree with the naming convention? I am not sure whether this is the right thing to do, and it seems it is breaking the "Check all files imported" test in CI. 
2. The last theorem is marked `@[ext]`, which the compiler complains about. I am unsure what to do about this.

**Comment by mattrobball** [comment]
The problematic lemma doesn't seem to fit the current vision for the `ext` tactic (what does it have to do with extensionality?) and so has been commented out.

**Comment by jcommelin** [comment]
For reviewing purposes, could someone please rewrite the git history a bit, so that

(1) a first commit moves the file to the correct folder
(2) the rest of the commits come after that one

**Comment by mo271** [comment]
Rebased and done as @jcommelin suggested.

**Comment by semorrison** [comment]
Hmm, I failed to make any further progress on the porting notes. Let's leave these for later.... :-(

bors d+

File: Mathlib/Algebra/Category/ModuleCat/Subobject.lean [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        · apply LinearMap.ext
+          intro x
+          rfl)
+      left_inv := fun N => by
+        --porting note: The type of `↾N.subtype` was ambiguous. Not entirely sure, I made the right
+        --choice here
+        convert congr_arg LinearMap.range
+            (underlyingIso_arrow (↾N.subtype : of R { x // x ∈ N } ⟶ M)) using 1
+        · have :
+            --porting note: added the `.toLinearEquiv.toLinearMap`
```
Comment (digama0) [inline Mathlib/Algebra/Category/ModuleCat/Subobject.lean:64]:
```suggestion
        -- Porting note: The type of `↾N.subtype` was ambiguous. Not entirely sure, I made the right
        -- choice here
        convert congr_arg LinearMap.range
            (underlyingIso_arrow (↾N.subtype : of R { x // x ∈ N } ⟶ M)) using 1
        · have :
            -- Porting note: added the `.toLinearEquiv.toLinearMap`
```

**Pull request closed without merging**